app/models/webhook.rb

```diff
@@ -0,0 +1,113 @@
+# == Schema Information
+#
+# Table name: webhooks
+#
+#  id             :integer          not null, primary key
+#  namespace_id   :integer
+#  url            :string(255)
+#  username       :string(255)
+#  password       :string(255)
+#  request_method :integer
+#  content_type   :integer
+#  enabled        :boolean          default("0")
+#  created_at     :datetime         not null
+#  updated_at     :datetime         not null
+#
+# Indexes
+#
+#  index_webhooks_on_namespace_id  (namespace_id)
+#
+
+require "base64"
+require "typhoeus"
+require "securerandom"
+require "json"
+
+class Webhook < ActiveRecord::Base
+  include PublicActivity::Common
+
+  enum request_method: ["GET", "POST"]
+  enum content_type: ["application/json", "application/x-www-form-urlencoded"]
+
+  belongs_to :namespace
+
+  has_many :deliveries, class_name: "WebhookDelivery"
+  has_many :headers, class_name: "WebhookHeader"
+
+  validates :url, presence: true
```

Comment:
Maybe it would be even better to validate whether the given URL is valid. A validation of this can perfectly be done with Ruby's stdlib like this (just googled for this :P):

```ruby
require 'uri'

begin
  uri = URI.parse(url)
  uri.kind_of?(URI::HTTP)
rescue URI::InvalidURIError
  false
end
```

This could be put in a validor (e.g. what we do with `NamespaceValidator`).